Update sharing-boards.rst (#5472)
mattermost/docs: source/boards/sharing-boards.rst

To share a board with others:

1. Open a board in any view.
2. On the top-right of the board select **Share board**.

> 3. Set the toggle on to **Publish to web and share this board with anyone**.

Reviewer: ```suggestion
3. Copy the internal link to **Share an editable view with users who have permissions to the board**.
-- OR --
4. Switch to the Publish tab and set the toggle on to **Publish and share a "read-only" link with everyone on the web**.
```

Reviewer: There's an extra "internal link" option in the Share modal. Maybe using numbered bullet points isn't the best with the "OR" option, but added details on the new functionality. Feel free to fix up the formatting as you see fit @justinegeffen.

> 4. Copy the link provided.

Reviewer: ```suggestion
5. Copy the link provided.
```

To share a board with others:

1. Open a board in any view.

> 2. On the top-right of the board select **Share board**.

Reviewer 2: ```suggestion
2. Select **Share board** in the top-right corner of the board.
```

3. The following two options are available:

>    * Copy the internal link to **Share an editable view with users who have permissions to the board**.

Reviewer 2: ```suggestion
   * On the **Share** tab, you can share an editable view link with users who have permissions to the board.
```

>    * On the **Publish** tab, you can share a read-only link online with everyone.

Reviewer 3: The "Enable Publicly-Shared Boards" must be turned on in the System Console for this option to be enabled (it's currently off by default). Just checking that this is covered somewhere already. Suggest linking to it if so.

>    * Switch to the **Publish** tab and set the toggle on to **Publish and share a "read-only" link with everyone on the web**.

Reviewer 2: ```suggestion
   * On the **Publish** tab, you can share a read-only link online with everyone.
```
